Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.swift

```diff
@@ -0,0 +1,185 @@
+                    .disabled(model.tracingActivity == .runningTrace)
+                    Spacer()
+                    Button("Trace") {
+                        Task { await model.trace() }
+                    }
+                    .disabled(!model.traceEnabled)
+                    Spacer()
+                    Button("Reset") {
+                        model.reset()
+                        Task { await mapViewProxy.setViewpointCenter(model.startingLocationPoint, scale: 3_000) }
+                    }
+                    .disabled(!model.resetEnabled)
+                }
+            }
+            .sheet(isPresented: $isConfigurationPresented) {
+                if #available(iOS 16, *) {
+                    NavigationStack {
+                        configurationView
+                    }
+                } else {
+                    NavigationView {
+                        configurationView
+                    }
+                }
+            }
+            .overlay(alignment: .center) {
+                if let tracingActivity = model.tracingActivity {
+                    VStack {
+                        Text(tracingActivity.label)
+                        ProgressView()
+                            .progressViewStyle(.circular)
+                    }
+                    .padding(6)
+                    .background(.thinMaterial)
+                    .cornerRadius(10)
+                }
+            }
+            .alert(
+                "Select Terminal",
+                isPresented: $model.terminalSelectorIsOpen,
+                actions: { terminalPickerButtons }
+            )
+        }
+    }
+    
+    /// Buttons for each the available terminals on the last added utility element.
+    @ViewBuilder var terminalPickerButtons: some View {
+        if let lastAddedElement = model.lastAddedElement,
+           let terminalConfiguration = lastAddedElement.assetType.terminalConfiguration {
+            ForEach(terminalConfiguration.terminals) { terminal in
+                Button(terminal.name) {
+                    lastAddedElement.terminal = terminal
+                    model.terminalSelectorIsOpen = false
+                    model.addTerminal(to: lastSingleTap!.mapPoint)
+                }
+            }
+        }
+    }
+    
+    @ViewBuilder var configurationView: some View {
+        Form {
+            Section {
+                List(model.filterBarrierCategories, id: \.name) { category in
+                    HStack {
+                        Text(category.name)
+                        Spacer()
+                        if category === model.selectedCategory {
+                            Image(systemName: "checkmark")
+                                .foregroundColor(.accentColor)
+                        }
+                    }
+                    // Allows the whole row to be tapped. Without this only the text is
+                    // tappable.
+                    .contentShape(Rectangle())
```

Comment:
I'm OK with the current UI. Curious if you have tried the inline picker here? I assume you used a list of HStack instead of a picker to allow nil-able selection.

Reply:
Yes I used a list so that the category can be unselected.